crates/oxc_ast_macros/src/lib.rs

```diff
@@ -53,3 +53,32 @@ pub fn ast(_args: TokenStream, input: TokenStream) -> TokenStream {
 pub fn ast_derive(_item: TokenStream) -> TokenStream {
     TokenStream::new()
 }
+
+/// Derive macro generating an impl of the trait `CloneIn`.
+///
+/// NOTE: This is an internal macro!
+/// # Panics
+///
+#[proc_macro_derive(CloneIn)]
+pub fn derive_clone_in(item: TokenStream) -> TokenStream {

```

Comment:
This macro only supports structs/enums with no generics/lifetimes and uses the std clone underneath. This is a temporary solution to avoid cross-crate code generation, but it can be a good way to allow deriving `CloneIn` without using codegen.
If necessary I can share the implementation between proc-macro and codegen so both `derive(CloneIn)` and `generate_derive(CloneIn)` output the same thing.

Reply:
Initially I didn't understand why we don't just use `#[generate_derive(CloneIn)]` for everything. But now I get it - the generated impls would have to go in the `oxc_span` and `oxc_syntax` crates, not `oxc_ast`.

Lucky none of the types in other crates have lifetimes!